### PR TITLE
Allow deleting multisig transactions not executed

### LINKED ISSUE #1757: Delete proposed not executed transactions
# What is needed?
Allow proposed transactions to be deleted by the proposer:
- A signature from the proposer will be required.
- Add a event for transaction deletion so they can be refreshed from client gateway caches (different issue, #1805)

A EIP712 will be signed, and epoch will be included to prevent replay attacks

# EIP712
```python
{
    "types": {
        "EIP712Domain": [
            {"name": "name", "type": "string"},
            {"name": "version", "type": "string"},
            {"name": "chainId", "type": "uint256"},
            {"name": "verifyingContract", "type": "address"},
        ],
        "DeleteRequest": [
            {"name": "safeTxHash", "type": "bytes32"},
            {"name": "totp", "type": "uint256"},
        ],
    },
    "primaryType": "DeleteRequest",
    "domain": {
        "name": "Safe Transaction Service",
        "version": "1.0",
        "chainId": chain_id,
        "verifyingContract": safe_address,
    },
    "message": {
        "safeTxHash": safe_tx_hash,
        "totp": totp,
    },
}
```

# Endpoint
`DELETE` `multisig-txs/<safe-tx-hash>/`

# Body
```json
{
  "timestamp": "<integer unix epoch>"
  "signature": "<0x hex string>"
}
```

Superseeds #1737

### PR DESCRIPTION
**Pull request by Uxio0**
- Closes #1757

### CODE REVIEW

File: safe_transaction_service/history/serializers.py [hunk 1]
```diff
@@ -472,6 +475,51 @@ def validate(self, attrs):
...
+            message_hash = DeleteMultisigTxSignatureHelper.calculate_hash(
+                safe_address, safe_tx_hash, chain_id, previous_totp=previous_totp
+            )
+            safe_signatures = SafeSignature.parse_signature(signature, message_hash)
+            if len(safe_signatures) != 1:
+                raise ValidationError(
+                    f"1 owner signature was expected, {len(safe_signatures)} received"
+                )
+            safe_signature = safe_signatures[0]
+            if safe_signature.signature_type != SafeSignatureType.EOA:
```
Comment (moisses89) [inline safe_transaction_service/history/serializers.py:515]: Which is the reason to only support EOA? maybe we could leave a comment here.

File: safe_transaction_service/history/serializers.py [hunk 2]
```diff
@@ -472,6 +475,51 @@ def validate(self, attrs):
         )
 
 
+class SafeMultisigTransactionDeleteSerializer(serializers.Serializer):
+    safe_tx_hash = Sha3HashField()
+    signature = HexadecimalField(min_length=65)
```
[the remark this answers is not in the document]
Comment (Uxio0) [reply safe_transaction_service/history/serializers.py:480]: We are not storing it on database, but I agree we should add a limit

File: safe_transaction_service/history/serializers.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/history/serializers.py:515]: If we want to suport more than EOA we would need to use the EIP712 domain of the Safe, so we would make it more difficult and I don't think there's a single person proposing transactions using different things from an EOA, I will add a comment

File: safe_transaction_service/history/serializers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Uxio0) [reply safe_transaction_service/history/serializers.py:480]: Added limit for signatures in this serializer and in others

File: safe_transaction_service/history/serializers.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/history/serializers.py:515]: Done

**Review by moisses89 — APPROVED** [review]

**Pull request merged**